### PR TITLE
Feature: Hoppity Purse Blocker

### PR DESCRIPTION
**Pull request by DavidArthurCole**
## What
Adds a blocker to prevent opening Hoppity's shop without coins.

## Changelog New Features
+ Added a feature to block opening Hoppity's trade menu from Abiphone calls if you do not have coins in your purse. - Daveed

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityCallWarning.kt [hunk 1]
```diff
@@ -128,6 +141,23 @@ object HoppityCallWarning {
...
+        if (!pickupOutgoingCommandPattern.matches(event.message)) return
+        if (commandSentTimer.passedSince() < 5.seconds) return
+        if (PurseAPI.getPurse() >= config.coinThreshold) return
+
+        commandSentTimer = SimpleTimeMark.now()
+        event.cancel()
+        ChatUtils.clickToActionOrDisable(
+            "§cBlocked picking up Hoppity without enough coins!",
+            config::ensureCoins,
+            actionName = "open bank menu",
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityCallWarning.kt:156]:
```suggestion
            actionName = "Open bank menu",
```

**Review by Thunderblade73 — COMMENTED** [review]
I don't see any problem in code (but I also don't understand why this exist or the description of the config is bad)

File: src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityCallWarning.kt (hunk 1 above)
Comment (DavidArthurCole) [reply src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityCallWarning.kt:156]: image

Comment (Thunderblade73) [reply src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityCallWarning.kt:156]: ok

**Comment by DavidArthurCole** [comment]
> I don't see any problem in code (but I also don't understand why this exist or the description of the config is bad)

Once you open the hoppity call (i.e., click Yes), if you don't have coins, you lose out on the offer - so, block opening the call if you don't have coins. Not sure if the config needs updating.

**Comment by Thunderblade73** [comment]
Yeah, that makes sense. But that isn't what the config reads to me. Since I thought it was about the normal hoppity shop, which didn't make sense.

**Review by Thunderblade73 — APPROVED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityCallWarning.kt [hunk 2]
```diff
@@ -128,6 +141,23 @@ object HoppityCallWarning {
         GlStateManager.color(1F, 1F, 1F, 1F)
     }
 
+    @SubscribeEvent
+    fun onCommandSend(event: MessageSendToServerEvent) {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityCallWarning.kt:145]: should have some sort of skyblock check

**Review by CalMWolfs — CHANGES_REQUESTED** [review]

**Review by CalMWolfs — APPROVED** [review]

**Pull request merged**